### ISSUE TITLE
Rule 7330 might be incorrectly diagnosed

### ISSUE THREAD

**Issue by dng-guillaume** [Trivadis/plsql-cop-sqldev#18]
Hello,

I am currently testing your plugin on SQLdev, and I generally love it. I don't fully agree with some rules but I can discard them, this is great.
While crawling old packages just to see what the report would find, I noticed that rule G-7330 (always assign values to OUT parameters) may be incorrectly raised sometimes.
In my case, the procedure puts data in the OUT parameter using a bulk collect into.

Here is an extract of the procedure : 
```
PROCEDURE get_calculation_data (
        val_run_id                  IN   asa_run_vw.run_id%TYPE,
        val_plate_id                IN   asa_plate_vw.plate_id%TYPE,
        val_calculation_meta_data   IN   rec_calculation_meta_data,
        var_calculation_data OUT NOCOPY tbl_calculation_data,
        val_is_control              IN   asa_well_type_lu.is_control%TYPE DEFAULT 'N'
    ) IS

        loc_current_date   asa_well_calc_data.created_date%TYPE := sysdate;
        loc_current_user   asa_well_calc_data.created_by%TYPE := nvl (
            apex_application.g_user,
            user
        );
    BEGIN
        SELECT
            vwd.well_id,
            vwd.this_value,
            vwd.read_index,
            val_calculation_meta_data.calc_alias ||
            '_' ||
            vwd.value_name,
            val_run_id,
            val_calculation_meta_data.calc_alias ||
            '_' ||
            vwd.value_name,
            asa_well_calc_data_sq.NEXTVAL,
            loc_current_user,
            loc_current_date,
            loc_current_user,
            loc_current_date
        BULK COLLECT
        INTO var_calculation_data
        FROM
            asa_valid_well_data_vw vwd
        WHERE
            vwd.value_name = val_calculation_meta_data.readout_name
            AND vwd.run_id     = val_run_id
            AND vwd.plate_id   = val_plate_id
            AND (val_is_control IS NULL
                 OR vwd.is_control  = val_is_control);

    END get_calculation_data;
```

I'm sure some other things could be said about the content of this extract, but right now I am just focusing on rule 7330 😁 

Of course, please let me know if I misunderstood something.

Cheers,

Guillaume

**Comment by PhilippSalvisberg** [comment]
Hi @dng-guillaume, thank you for reporting this bug.

I can reproduce the behaviour with the following code (your code, just added a package body to make it valid):

```sql
CREATE OR REPLACE PACKAGE BODY demo AS
   PROCEDURE get_calculation_data (
        val_run_id                  IN   asa_run_vw.run_id%TYPE,
        val_plate_id                IN   asa_plate_vw.plate_id%TYPE,
        val_calculation_meta_data   IN   rec_calculation_meta_data,
        var_calculation_data OUT NOCOPY tbl_calculation_data,
        val_is_control              IN   asa_well_type_lu.is_control%TYPE DEFAULT 'N'
    ) IS

        loc_current_date   asa_well_calc_data.created_date%TYPE := sysdate;
        loc_current_user   asa_well_calc_data.created_by%TYPE := nvl (
            apex_application.g_user,
            user
        );
    BEGIN
        SELECT
            vwd.well_id,
            vwd.this_value,
            vwd.read_index,
            val_calculation_meta_data.calc_alias ||
            '_' ||
            vwd.value_name,
            val_run_id,
            val_calculation_meta_data.calc_alias ||
            '_' ||
            vwd.value_name,
            asa_well_calc_data_sq.NEXTVAL,
            loc_current_user,
            loc_current_date,
            loc_current_user,
            loc_current_date
        BULK COLLECT
        INTO var_calculation_data
        FROM
            asa_valid_well_data_vw vwd
        WHERE
            vwd.value_name = val_calculation_meta_data.readout_name
            AND vwd.run_id     = val_run_id
            AND vwd.plate_id   = val_plate_id
            AND (val_is_control IS NULL
                 OR vwd.is_control  = val_is_control);

    END get_calculation_data;
END demo;
/
```

A "G-7330: Always assign values to OUT parameters." violation is reported for line 6. However, the parameter `var_calculation_data` was populated in line 33 using `bulk collect`. So this is clearly a bug.

**Comment by PhilippSalvisberg** [comment]
fixed with db* CODECOP v4.4 (commit c621fd786521480c4af3c3acb9cae34e6d197032)
